File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.py
class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        freq = {}
        maxi=0
        maxe=0
        for num in nums:
            freq[num] = freq.get(num, 0)+1
        
        for val in freq.values():
            maxi = max(maxi, val)
        
        for val in freq.values():
            if val == maxi:
                maxe+=1
        
        return maxe*maxi
        

File: 3242-count-elements-with-maximum-frequency/count-elements-with-maximum-frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> freq;
        for(int num : nums) freq[num]++;
        int maxi = 0;
        for(auto& fre : freq){
            maxi = max(maxi, fre.second);
        }
        int maxe = 0;
        for(auto& fre : freq){
            if(fre.second == maxi) maxe++;
        }
        return maxe*maxi;
    }
};